pennylane/gradients/finite_difference.py

```diff
@@ -724,13 +724,13 @@ def processing_fn(results):
 
             # compute the linear combination of results and coefficients
             res = qml.math.stack(res)
-            g = sum(c * r for c, r in zip(coeffs, res))
+            g = sum(r * c for c, r in zip(coeffs, res))

```

Comment:
This makes it possible to use a data type for the results that does not define `__rmul__`